### DISCUSSION TITLE
K8s cluster switchover

### DISCUSSION THREAD

**Discussion by aryklein** [cloudnative-pg/cloudnative-pg#324] [Q&A] [answered]
Does the operator support cluster switchover In a multi-cluster deployment? Let's say I want to promote the replica K8s cluster as primary k8s PostgreSQL cluster:

![2022-06-28_13-43-46](https://user-images.githubusercontent.com/8699850/176235462-b82d330c-3ef6-4c14-948e-34e3d54f95be.png)

I wasn't able to find documentation on this use case

**Comment by gbartolini** [comment]
Hi, thanks for starting this.

This is one of my favorite capabilities. We have called this feature “Physical Replica Clusters”. They are nothing but read clusters, usually in a different Kubernetes cluster hosted in a different region, and they represent an efficient way to achieve very good business continuity results through disaster recovery.

The operator provides the primitives for you to control the promotion of a replica cluster, but at the moment it doesn’t work across kubernetes clusters. Kubernetes federation will help address this in the future.

Out of curiosity, have you looked at this section in the docs?

https://github.com/cloudnative-pg/cloudnative-pg/blob/main/docs/src/replica_cluster.md

**Reply by aryklein** [reply]
Thanks @gbartolini for answering.

I was looking for some feature like `repmgr standby switchover` but at cluster level. In other words, to promote the former primary cluster as a replica following the new primary cluster without recreating a new postgresql cluster, but I just read in the link you sent me this:

> !!! Note Disabling replication is an irreversible operation: once replication is disabled and the designated primary is promoted to primary, the replica cluster and the source cluster will become two independent clusters definitively.

Do you have any idea if there is a hack that we could do to accomplish this feature outside of the operatoror boundaries? 

Thanks in advance,
Ary

**Reply by allamand** [reply]
I would definitely want a solution at cluster level so that I would be able to upgrade my Kubernetes clusters in a blue/green fashion. Is this something that has evolve since the previous response ?

**Reply by gbartolini** [reply, marked as answer]
It is already available.

See: https://cloudnative-pg.io/documentation/current/replica_cluster/#distributed-topology
